static/w18/hw/c4cs-wk11-homework.tex

```diff
@@ -0,0 +1,285 @@
+\documentclass{article}
+\usepackage[T1]{fontenc}
+
+\usepackage{amssymb}
+\usepackage{courier}
+\usepackage{fancyhdr}
+\usepackage{fancyvrb}
+\usepackage[top=.75in, bottom=.75in, left=.75in,right=.75in]{geometry}
+\usepackage{graphicx}
+\usepackage{lastpage}
+\usepackage{listings}
+\lstset{basicstyle=\small\ttfamily}
+\usepackage{mdframed}
+\usepackage{parskip}
+\usepackage{soul}
+\usepackage{tabularx}
+\usepackage{textcomp}
+\usepackage{upquote}
+\usepackage{xcolor}
+
+% http://www.monperrus.net/martin/copy-pastable-ascii-characters-with-pdftex-pdflatex
+\lstset{
+upquote=true,
+columns=fullflexible,
+keepspaces=true,
+literate={*}{{\char42}}1
+         {-}{{\char45}}1
+         {^}{{\char94}}1
+}
+\lstset{
+  moredelim=**[is][\color{blue}\bf\small\ttfamily]{@}{@},
+}
+
+% http://tex.stackexchange.com/questions/40863/parskip-inserts-extra-space-after-floats-and-listings
+\lstset{aboveskip=6pt plus 2pt minus 2pt, belowskip=-4pt plus 2pt minus 2pt}
+
+
+
+\usepackage[colorlinks,urlcolor={blue}]{hyperref}
+\usepackage[capitalise,nameinlink,noabbrev]{cleveref}
+\crefname{section}{Question}{Questions}
+\Crefname{section}{Question}{Questions}
+
+\begin{document}
+
+\fancyhead[C]{\hl{Select the right page in Gradescope or we will not grade the question!}}
+\fancyhead[L]{}
+\fancyhead[R]{}
+
+\fancyfoot[L]{\color{gray} C4CS -- W'18}
+\fancyfoot[R]{\color{gray} Revision 1.0}
+\fancyfoot[C]{\color{gray} \thepage~/~\pageref*{LastPage}}
+\pagestyle{fancyplain}
+
+\title{\textbf{Homework 11\\Debugging}}
+\author{\textbf{\color{red}{Due: Wednesday, March 27th, 11:59PM (Hard Deadline)}}}
+\date{}
+\maketitle
+
+
+\section*{Submission Instructions}
+Submit this assignment on \href{https://gradescope.com/courses/13649}{Gradescope}.
+You may find the free online tool \href{https://www.pdfescape.com}{PDFescape}
+helpful to edit and fill out this PDF.
+You may also print, handwrite, and scan this assignment.
+
+\newpage
+\section{\texttt{Git}ting started}
+
+Please \texttt{clone} the following repository:
+
+\begin{lstlisting}
+@>@ git clone https://github.com/c4cs/debugging-basics.git
+\end{lstlisting}
+
+This repository contains 4 branches: \texttt{master}, \texttt{gdb-debug-1}, \texttt{gdb-debug-2}, and \texttt{valgrind-debug}. Each part of this homework will take place on a different branch.
+
+\section{Finding primes}
+
+The repository you have cloned is an attempt at a program to find prime
+numbers. It will search from 3 up to a maximum number input by the user.
+
+The program intends to follow the structure:
+\begin{lstlisting}
+  o prompt user for upper bound
+  o for N=3..upper bound:
+    - check if N is prime
+      * if no prime n between 1..sqrt(N) divides N, then prime
+    - save whether N is prime for future loops to use
+    - if N is prime, print N
+\end{lstlisting}
+
+\subsection{Debugging with \texttt{gdb}}
+
+\emph{You may find it helpful to consult the
+  \href{https://c4cs.github.io/lectures/f17/week11}{gdb lecture notes}
```

Comment:
I think this link needs to be updated to point to this semester's lecture slides

Reply:
Fixed with commit 83d1ef0